sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -414,7 +449,7 @@ class SpillableDeviceBufferHandle private (
 
   override val approxSizeInBytes: Long = sizeInBytes
 
-  private[spill] override def spillable: Boolean = synchronized {
+  private[spill] override def spillable: Boolean = {
     if (super.spillable) {
       dev.getOrElse {
```

Comment:
The only thing I can think of is if a handle can be closed outside of a spilling context, and if so, could this happen where multiple threads hold a ref to the handle? Because I see that `close` releases the device resources, but not sure when exactly when that may happen. @abellina do you know if this would be safe?